Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/extrude.h

```diff
@@ -0,0 +1,316 @@
+// Copyright (c) 2018 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+// You can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0+
+//
+//
+// Author(s)     : Sebastien Loriot, Maxime Gimeno
+
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_EXTRUDE_H
+#define CGAL_POLYGON_MESH_PROCESSING_EXTRUDE_H
+
+#include <CGAL/license/Polygon_mesh_processing/meshing_hole_filling.h>
+
+
+#include <CGAL/Polygon_mesh_processing/orientation.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+#include <CGAL/boost/graph/named_function_params.h>
+#include <CGAL/boost/graph/copy_face_graph.h>
+#include <CGAL/Kernel_traits.h>
+#include <CGAL/boost/graph/Euler_operations.h>
+#include <vector>
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+namespace extrude_impl{
+
+template<typename PMAP, typename Vector>
+struct Const_dist_translation{
+  Const_dist_translation(PMAP map, const Vector& dir)
+    :map(map), dir(dir){}
+  
+  template<typename VertexDescriptor, typename U>
+  void operator()(const VertexDescriptor vd, const U&)
+  {
+    typename boost::property_traits<PMAP>::value_type p = get(map, vd) + dir;
+    put(map, vd, p);
+  }
+  
+  PMAP map;
+  Vector dir;
+};
+
+struct Identity_functor
+{
+  template<typename T, typename U>
+  void operator()(const T&, const U&){}
+};
+}//end extrude_impl
+
+/**
+ * \ingroup PMP_meshing_grp
+ * \brief this overload performs a generalized extrusion of `input`.
+ * 
+ * extrudes the open surface mesh `input` and puts the result in `output`. The mesh generated is a closed 
+ * surface mesh with a bottom and top part, both having the same graph combinatorics as `input` (except 
+ * that the orientation of the faces of the bottom part is reversed). The bottom and the top parts are 
+ * connected by a triangle strip between boundary cycles. The coordinates of the points associated to the 
+ * vertices of the bottom and top part are first initialized to the same value as the corresponding 
+ * vertices of `input`. Then for each vertex, a call to `bot` and `top` is done for the vertices of the 
+ * bottom part and the top part, respectively.
+ * @tparam InputMesh a model of `FaceListGraph` and `MutableFaceGraph`
```

Comment:
I don't think `InputMesh` should be mutable (it is `const` in the function signature). Right?